src/typemap/src/network_helpers.rs

```diff
@@ -0,0 +1,133 @@
+//! Network-related helpers
+//!
+//! This module provides helpers to translate a guest-provided sockaddr buffer into a 
+//! host-usable pointer and to compute the correct socklen_t for Linux. It is used by 
+//! our socket-related syscalls to bridge from per-cage virtual memory to host libc calls.
+use sysdefs::data::net_struct::SockAddr;
+use libc::{sockaddr, strlen};
+use sysdefs::constants::lind_platform_const::LIND_ROOT;
+use sysdefs::constants::net_const::{AF_UNIX};
+use cage::{get_cage, memory::memory::translate_vmmap_addr};
+use crate::datatype_conversion::validate_cageid;
+use std::os::raw::{c_void, c_char};
+use std::ptr;
+
+/// `unix_len_from_sun_path` computes the effective `socklen_t` for an `AF_UNIX` 
```

Comment:
It may be fine but I've been staring at this function even after reading the comments and having a pretty difficult time parsing it, may be better to have some of the comments inline.